extensions/webstorage/lib.rs

```diff
@@ -51,82 +46,59 @@ pub fn get_declaration() -> PathBuf {
   PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("lib.deno_webstorage.d.ts")
 }
 
-struct WebStorageConnectionResource(Connection);
-
-impl Resource for WebStorageConnectionResource {
-  fn name(&self) -> Cow<str> {
-    "webStorage".into()
-  }
-}
-
-pub fn op_webstorage_open(
+fn get_webstorage(
   state: &mut OpState,
   persistent: bool,
-  _zero_copy: Option<ZeroCopyBuf>,
-) -> Result<u32, AnyError> {
-  let connection = if persistent {
-    let path = state.try_borrow::<LocationDataDir>().ok_or_else(|| {
-      DomExceptionNotSupportedError::new(
-        "LocalStorage is not supported in this context.",
-      )
-    })?;
-    std::fs::create_dir_all(&path.0)?;
-    Connection::open(path.0.join("local_storage"))?
-  } else {
-    Connection::open_in_memory()?
-  };
-
-  connection.execute(
-    "CREATE TABLE IF NOT EXISTS data (key VARCHAR UNIQUE, value VARCHAR)",
-    params![],
-  )?;
+) -> Result<&Connection, AnyError> {
+  if state.try_borrow::<Connection>().is_none() {
```

Comment:
I think you'll need two wrapper structs: `struct LocalStorage(Connection)` and `struct SessionStorage(Connection)`; otherwise you will have only a single instance in `OpState` that will depend which API was used first